### ISSUE TITLE
Wrong result with `date2num`

### ISSUE THREAD

**Issue by shaharkadmiel** [Unidata/cftime#140]
`cftime.date2num` is returning a wrong result. See example below:

```python
import cftime as cf
import pandas as pd

print('cftime version:', cf.__version__)

reftime = pd.to_datetime('2018-01-23 09:31:42.94')
t = pd.to_datetime('2018-01-23 09:27:10.950000')

# seconds since a reference time
cf_delta = cf.date2num(t.to_pydatetime(), f'seconds since {reftime}')

print('cftime date2num:', cf_delta)
```

Output:
```python
cftime version: 1.0.4.2
cftime date2num: -271.05
```
The answer should be:

```python
# using pandas timedelta calculation
pd_delta = (t - reftime).total_seconds()
print('pandas timedelta:', pd_delta)

# manual calculation
delta = (
    (t.minute * 60 + t.second + t.microsecond * 1e-6) -
    (reftime.minute * 60 + reftime.second + reftime.microsecond * 1e-6)
)
print('manual timedelta:', delta)
```

Output:
```python
pandas timedelta: -271.99
manual timedelta: -271.99
```

**Comment by jswhit2** [comment]
cftime cannot currently handle fractional seconds in the reference date.

**Comment by shaharkadmiel** [comment]
That is unfortunate...

**Comment by jswhit2** [comment]
It's a bug - it should work.

**Comment by shaharkadmiel** [comment]
I belive the bug is here: https://github.com/Unidata/cftime/blob/ec52b8779c5e55994ef2232f9212bb42c5b01803/cftime/_cftime.pyx#L226

Even python datetime arithmetic can handle this:
```python
delta = t.to_pydatetime() - reftime.to_pydatetime()
print(type(delta), delta.total_seconds())
```

Output:
```python
<class 'datetime.timedelta'> -271.99
```

**Comment by jswhit2** [comment]
It's a simple fix, PR coming...

**Comment by jswhit** [comment]
Fixed by PR #141  - works now in master and will be in 1.1.0 release.
